oak_functions/loader/src/server.rs

```diff
@@ -153,15 +153,26 @@ pub trait ExtensionFactory {
 /// `Uwabi` extension called by listening to a channel.
 pub enum BoxedExtension {
     Native(Box<dyn OakApiNativeExtension + Send + Sync>),
-    Uwabi(Box<dyn UwabiExtension + Send + Sync>),
+    Uwabi(BoxedUwabiExtension),
 }
 
+pub type BoxedUwabiExtension = Box<dyn UwabiExtension + Send + Sync>;
+
 pub type BoxedExtensionFactory = Box<dyn ExtensionFactory + Send + Sync>;
 
 /// Trait for implementing an extension which relies on UWABI.
 pub trait UwabiExtension {
     /// Get the channel handle to address this extension.
     fn get_channel_handle(&self) -> ChannelHandle;
+
+    /// Get the endpoint.
+    fn get_endpoint(&mut self) -> Option<&mut Endpoint>;
+
+    /// Set the endpoint if it has not been set before.
+    // Note: we cannot set the endpoint when we `create` the extension, as this would require
+    // to change the `BoxedExtensionFactory` trait. This helps to keep the changes to the
+    // (existing) Native extensions minimal.
+    fn set_endpoint(&mut self, endpoint: Endpoint);
```

Comment:
This is going to be cleaned up when all the extensions are migrated, right?

Reply:
Yes.

Reply:
#2510